pylint/extensions/while.py

```diff
@@ -0,0 +1,30 @@
+"""Check for use of while loops."""
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+
+
+class WhileChecker(BaseChecker):
+
+    __implements__ = (IAstroidChecker,)
+    name = "while_used"
+    msgs = {
+        "W0149": (
+            "Used while loop",
+            "while-used",
+            "Unbounded `while` loops can almost always be rewritten as bounded `for` loops.",
+        )
+    }
+
+    @check_messages("while-used")
+    def visit_while(self, node):
+        self.add_message("while-used", node=node)
+
+
+def register(linter):
+    """Required method to auto register this checker.
+
+    :param linter: Main interface object for Pylint plugins
+    :type linter: Pylint object
+    """
+    linter.register_checker(WhileChecker(linter))
```

Comment:
We should not register it, or it become a default checker and we don't want that :) But we can add a functional test where the plugin is loaded to check it.

Reply:
I think the fact that it is in `extensions` keeps it from being a default.

In any case, removing the register line causes `AttributeError: module 'pylint.extensions.while_used' has no attribute 'register'`.

Reply:
Sorry, my bad :)